### PR TITLE
update development tools

### PR DESCRIPTION
**Pull request by sdemos**
update the tooling we use here to use glide instead of godeps and use a makefile instead of random build scripts.

### CODE REVIEW

**Comment by sdemos** [comment]
yeah I just noticed that too. I was testing this with a dirty git tree (I had already regenerated the bindings for the new api changes coming down the pipe). I'm fixing it now.

**Comment by sdemos** [comment]
I'm also moving the `go build` command to `all` so that it just reruns it every time instead of having to call clean every time you want to compile new changes.

**Comment by dgonyeo** [comment]
Or you could do something like Euan did for ct: https://github.com/coreos/container-linux-config-transpiler/commit/e8a59e9d7b6d6e890a8be71bc52f1547886023a2

Either way.

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,19 @@
+.PHONY:	all clean test vendor
```
Comment (csssuf) [inline Makefile]: Should `api-gen` be a PHONY target too?

**Comment by sdemos** [comment]
eh, it doesn't really matter, the build is pretty quick anyway. just seems like a hassle to tell make when to rebuild it if there is only one build step anyway.

File: Makefile (hunk 1 above)
Comment (sdemos) [reply Makefile]: missed that, thanks

**Comment by sdemos** [comment]
@dgonyeo @csssuf updated

**Comment by sdemos** [comment]
the changes that came from regenerating the bindings came from a few places - 
* it's been literally years since those were generated
* I updated the bindings manually for the update percent changes and apparently I was a little off of what it actually would've generated

**Pull request merged**